File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int n = nums.length;
        int maxVal = (1 << maximumBit) - 1;
        int xorAll = 0;
        for (int x : nums) xorAll ^= x;
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            ans[i] = maxVal ^ xorAll;
            xorAll ^= nums[n - 1 - i];
        }
        return ans;
    }
}

File: LeetCode/1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        n = len(nums)
        maxVal = (1 << maximumBit) - 1
        xorAll = 0
        for x in nums:
            xorAll ^= x
        ans = []
        for i in range(n):
            ans.append(maxVal ^ xorAll)
            xorAll ^= nums[n - i - 1]
        return ans